crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -93,23 +93,23 @@ fn determine_rootdir<P: AsRef<Path>>(bundle: P, namespace: String) -> Result<Pat
 }
 
 impl Instance for Wasi {
-    type E = wasmtime::Engine;
-    fn new(id: String, cfg: Option<&InstanceConfig<Self::E>>) -> Self {
+    fn new(id: String, cfg: Option<&InstanceConfig>) -> Self {
         // TODO: there are failure cases e.x. parsing cfg, loading spec, etc.
         // thus should make `new` return `Result<Self, Error>` instead of `Self`
         log::info!("creating new instance: {}", id);
         let cfg = cfg.unwrap();
         let bundle = cfg.get_bundle().unwrap_or_default();
         let rootdir = determine_rootdir(bundle.as_str(), cfg.get_namespace()).unwrap();
+        let engine = Engine::default();
         Wasi {
             id,
             exit_code: Arc::new((Mutex::new(None), Condvar::new())),
-            engine: cfg.get_engine(),
             stdin: cfg.get_stdin().unwrap_or_default(),
             stdout: cfg.get_stdout().unwrap_or_default(),
             stderr: cfg.get_stderr().unwrap_or_default(),
             bundle,
             rootdir,
+            engine,
```

Comment:
is this needed here? The wasm-edge shim does the engine instance at the time of the execution and it looks like that could be done here as well.  Is there a perf benefit to doing it up front before execution starts?